Tools/AP_Periph/rc_out.cpp

```diff
@@ -63,6 +63,8 @@ void AP_Periph_FW::rcout_init()
     // run this once and at 1Hz to configure aux and esc ranges
     rcout_init_1Hz();
 
+    hal.rcout->set_dshot_esc_type(SRV_Channels::get_dshot_esc_type());
```

Comment:
```suggestion
#if HAL_DSHOT_ENABLED
    hal.rcout->set_dshot_esc_type(SRV_Channels::get_dshot_esc_type());
#endif
```

this same define guards the code